flink-runtime/src/main/java/org/apache/flink/runtime/executiongraph/Execution.java

```diff
@@ -402,21 +405,53 @@ public void onComplete(Throwable failure, Object success) throws Throwable {
 		}
 	}
 
+	public void stop() {
+		// sends stop RPC call
+
+		final SimpleSlot slot = this.assignedResource;
+
+		if (slot != null) {
+			final ActorGateway gateway = slot.getInstance().getActorGateway();
+
+			Future<Object> stopResult = gateway.retry(
+				new StopTask(attemptId),
+				NUM_STOP_CALL_TRIES,
+				timeout,
+				executionContext);
+
+			stopResult.onComplete(new OnComplete<Object>() {
+				@Override
+				public void onComplete(Throwable failure, Object success) throws Throwable {
+					if (failure != null) {
+						fail(new Exception("Task could not be stopped.", failure));
+					} else {
+						TaskOperationResult result = (TaskOperationResult) success;
+						if (!result.success()) {
+							LOG.debug("Stopping task call did not find task. Probably akka message call race.");
```

Comment:
Same pattern as `cancel` `onComplete` back-call (see line 901) -- I we change it, we should change both. I would keep it. If you want to get it changed for both, let me know.